Assignment4.py

```diff
@@ -0,0 +1,263 @@
+import pandas as pd
+import plotly.express as px
+import statsmodels
+from matplotlib import pyplot as plt
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import OneHotEncoder
+
+df = pd.read_csv(
+    "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/titanic.csv"
+)
+# print(df)
+
+df.rename(columns={"class": "v_class"}, inplace=True)
+# print(list(df))
+
+# make a list of predictor and response variables
+responses = ["survived"]
+predictors = [
+    "pclass",
+    "sex",
+    "age",
+    "sibsp",
+    "parch",
+    "fare",
+    "embarked",
+    "v_class",
+    "who",
+    "adult_male",
+    "deck",
+    "embark_town",
+    "alive",
+    "alone",
+]
+
+# print(list(responses))
+# print(list(predictors))
+
+# getting rid of missed values ASAP
+for col in df.columns:
+    if df[col].dtypes == "float":
+        df[col].fillna((df[col].mean()), inplace=True)
+    else:
+        df = df.apply(lambda col: col.fillna(col.value_counts().index[0]))
+
+# determine if boolean or continuous
+
+boolean_labels = {}
+for i in responses:
+    if df[i].nunique() == 2:
+        df[i] = df[i].astype("bool")
+        df.replace({False: 0, True: 1}, inplace=True)
+        boolean_labels = {idx: value for idx, value in enumerate(df[i].unique())}
+# print(boolean_labels)
+
+# create dictionary for predictors type
+predictors_type = {"continuous": [], "categorical": []}
+continuous = df.select_dtypes(include=["float"])
+
+for i in predictors:
```

Comment:
Bita your code logic is good. I think what you need to do now write a function so that professor can input any dataset into it, and then the function can call your logic so that it can separate out the categorical and continuous columns and then run them from those functions that the professor previously made on the graphs.

Reply:
also remove the n = 200 from your code as well from the functions that make the graphs. They don't mean anything.